app.js

```diff
@@ -285,20 +285,52 @@ window.onload = () => {
 
 /* Validate form */
 function onSubmit(event) {
-  const validate = document.getElementById('email');
+  function saveToLocalStorage(n, c, e) {
+    const data = { name: n, comment: c, email: e };
+    const storeData = JSON.stringify(data);
+    localStorage.setItem('data', storeData);
+  }
+
+  const formEmail = document.getElementById('email');
+  const formName = document.getElementById('name');
+  const formComment = document.getElementById('textarea');
   const errorMessage = document.getElementById('error');
-  const email = validate.value;
+  const name = formName.value;
+  const comment = formComment.value;
+  const email = formEmail.value;
 
   if (email.toLowerCase() !== email) {
     event.preventDefault();
-    validate.classList.add('invalid');
+    formEmail.classList.add('invalid');
     errorMessage.classList.add('error');
     errorMessage.innerText = 'Email should be lowercase!';
   } else {
-    validate.classList.remove('invalid');
+    formEmail.classList.remove('invalid');
     errorMessage.classList.remove('error');
+    saveToLocalStorage(name, comment, email);
+  }
+
+  // Load data
+  const loadData = JSON.parse(localStorage.getItem('storeData'));
+  if (loadData !== null) {
+    formName.value = loadData.name;
+    formComment.value = loadData.comment;
+    formEmail.value = loadData.email;
   }
+
+  // Reset form after sending the values
+  const reset = () => {
+    formName.value = '';
+    formComment.value = '';
+    formEmail.value = '';
+    localStorage.clear();
+  };
+
+  const buttonReset = document.getElementById('reset');
+  buttonReset.addEventListener('click', reset);
 }
 
 const form = document.getElementById('form');
-form.addEventListener('submit', onSubmit);
\ No newline at end of file
+form.addEventListener('submit', onSubmit);
```

Comment:
- [ ] Great job on implementing the localstorage :clap: , However we need to store the form data as soon as it is filled, currently you save the data only when the user submits the form, you an use the change event listener to save data to local storage once the form fields are filled.
- [ ] Also Please make sure when there is any form data in the local storage the form is populated with that data on page reload, you can use the reload event listener and check the local storage and if there is any data available populate your form with the appropriate data.
```
window.onload = () => {
  let savedFormData = localStorage.getItem('contactFormInput');
  savedFormData = JSON.parse(savedFormData);
  // Check if the form data object is found on localStorage
  if (savedFormData) {
  // populate inputs values if data was found
  // ex: nameInput.value = savedFormData.name
  }
};
```
The points to consider : 
1. The contact form information is persistent to page reloads.
2. After filling the form, and not submitting it, the entered values should be saved in the browser.
3. When visiting the page the form inputs should be contained, be auto-populated with, the values entered in the previous visit to the page.
4. If at the previous visit to the page, in the same browser, no input was entered in the form then the form inputs should remain empty.